cli/cmd/install.go

```diff
@@ -248,7 +251,34 @@ func newCmdInstall() *cobra.Command {
 	return cmd
 }
 
-func (options *installOptions) validateAndBuild(flags *pflag.FlagSet) (*installValues, *pb.All, error) {
+func (options *installOptions) validateAndBuild(args []string, flags *pflag.FlagSet) (*installValues, *pb.All, error) {
+	if len(args) > 1 {
+		return nil, nil, fmt.Errorf("only zero or one argument permitted, received: %s", args)
+	}
+	stage := ""
+	if len(args) == 1 {
+		// this is guaranteed to be a valid stage via `cobra.OnlyValidArgs`
+		stage = args[0]
+	}
+
+	// only a few flags are available for config stage
+	if stage == configStage && flags != nil {
+		var err error
+		flags.VisitAll(func(f *pflag.Flag) {
+			if f.Changed {
+				switch f.Name {
+				// TODO: remove "proxy-auto-inject" when it becomes default
+				case "linkerd-namespace", "proxy-auto-inject":
```

Comment:
I think we should add to this list `ignore-cluster` and all the global config flags.

Reply:
I meant all the "global flags" (--api-addr, etc)

Reply:
The `flags` var passed into this function comes from `installOptions.recordableFlagSet()`, which does not include the "global flags".

```bash
$ bin/go-run cli install config --api-addr foo > /dev/null && echo $?
0
$ bin/go-run cli install config --admin-port 123 > /dev/null && echo $?
Error: flag not available for config stage: --admin-port
Usage:
```

... that said, i'm realizing that there is also `installOnlyFlagSet`, none of which should be set during `linkerd install config`, i have pushed a new commit to reflect this. i don't think `linkerd install config --ignore-cluster` should be supported, as it has no effect on the output.

Reply:
> `linkerd install config --ignore-cluster` should be supported, as it has no effect on the output.

`linkerd install ` and `linkerd install config` both check if linkerd already exists in the cluster, so I was thinking on allowing that flag in both cases for consistency.

Reply:
oops sorry i mean `linkerd install config --ignore-cluster` should _not_ be supported (i think you read it the way i intended anyway ;) ) ...

nice catch! i failed to realize that `linkerd install config` would error out if a cluster already existed, and not providing an `--ignore-cluster` flag leaves the user completely unable to render that output.